### DISCUSSION TITLE
Does Ultralytics uses Gpu if GPU is available

### DISCUSSION THREAD

**Discussion by nuhmanpk** [ultralytics/ultralytics#2643] [Q&A] [answered]
Im running on a gpu enabled instance. Do I need to move the image to GPU before inference and when i'm printing the result 
```python
image = cv2.imread('./img.jpg')
result = model.predict(image)
print(result)
```
this is what the output is 

```
0: 480x640 1 ABC, 304.4ms
Speed: 2.8ms preprocess, 304.4ms inference, 8.7ms postprocess per image at shape (1, 3, 640, 640)
[{ '_keys': ('boxes', 'masks', 'probs'),
  'boxes': ultralytics.yolo.engine.results.Boxes
type:  torch.Tensor
shape: torch.Size([1, 6])
dtype: torch.float32
tensor([[  4.63111, 243.60742, 367.70370, 275.31027,   0.84515,   0.00000]]),
  'masks': None,
  'names': {0: 'ABC'},
  'orig_img': array([[[31, 47, 64],
        [30, 47, 63],
        [31, 47, 62],
        ...,
        [56, 60, 74],
        [64, 66, 79],
        [67, 67, 79]],

       [[31, 45, 62],
        [32, 47, 65],
        [32, 46, 61],
        ...,
        [34, 43, 57],
        [42, 49, 63],
        [54, 57, 71]],

       [[26, 41, 57],
        [24, 39, 55],
        [29, 43, 61],
        ...,
        [30, 35, 50],
        [31, 37, 52],
        [33, 41, 56]],

       ...,

       [[ 1,  1,  1],
        [ 1,  1,  1],
        [ 1,  1,  1],
        ...,
        [ 0,  0,  0],
        [ 0,  0,  0],
        [ 0,  0,  0]],

       [[ 1,  1,  1],
        [ 1,  1,  1],
        [ 1,  1,  1],
        ...,
        [ 0,  0,  0],
        [ 0,  0,  0],
        [ 0,  0,  0]],

       [[ 1,  1,  1],
        [ 1,  1,  0],
        [ 1,  1,  0],
        ...,
        [ 0,  0,  0],
        [ 0,  0,  0],
        [ 0,  0,  0]]], dtype=uint8),
  'orig_shape': (290, 399),
  'path': 'img.jpg',
  'probs': None,
  'speed': {'inference': 304.4161796569824, 'postprocess': 8.677959442138672, 'preprocess': 2.775907516479492}}]
```

what does These represent ?
```
type:  torch.Tensor
shape: torch.Size([1, 6])
dtype: torch.float32
tensor([[  4.63111, 243.60742, 367.70370, 275.31027,   0.84515,   0.00000]]),
```

**Comment by glenn-jocher** [comment, marked as answer]
@nuhmanpk hello! Yes, Ultralytics YOLOv8 will automatically use a GPU if it's available and CUDA is properly installed on your system. You don't need to manually move the image to the GPU before inference; the model's `predict` method handles that for you.

Regarding the output you're seeing, it represents the detection result for your image. Here's a brief explanation of the components:

- `type: torch.Tensor`: Indicates that the boxes are stored in a PyTorch tensor.
- `shape: torch.Size([1, 6])`: The shape of the tensor, where 1 indicates the number of detections and 6 represents the attributes of each detection.
- `dtype: torch.float32`: The data type of the tensor elements, which is 32-bit floating-point.
- `tensor([[  4.63111, 243.60742, 367.70370, 275.31027,   0.84515,   0.00000]]),`: The actual tensor values, which include:
  - The first four numbers are the coordinates of the bounding box (x1, y1, x2, y2).
  - The fifth number is the confidence score of the detection.
  - The last number is the class ID of the detected object (in this case, '0' which corresponds to 'ABC' as per the 'names' dictionary).

The rest of the output provides additional information such as the original image, the path to the image file, and the inference speed.

For more detailed explanations on the output format and how to interpret it, please refer to our documentation at https://docs.ultralytics.com. 😊
